_posts/2018-11-16-grpc-dotnet-build.md

```diff
@@ -0,0 +1,175 @@
+---
+layout: post
+title: gRPC Meets .NET SDK And Visual Studio
+published: true
+permalink: blog/grpc-dotnet-build
+author: Kirill 'kkm' Katsnelson
+author-link: https://github.com/kkm000
+company: SmartAction
+company-link: https://www.smartaction.ai
+---
+As part of Microsoft's move towards its cross-platform .NET offering, they have
+greatly simplified the project file format, and allowed a tight integration of
+third-party code generators with .NET projects. We are listening, and now proud
+to introduce integrated compilation of Protocol Buffer and gRPC service
+`.proto` files in .NET C# projects starting with the version 1.17 of the
+Grpc.Tools NuGet package, now available from Nuget.org.
+
+<!--more-->
+
+## A Walkthrough
+
+Now you can include `.proto` files along with C# source files into `.csproj`
+project files. In this blog post, we'll walk through the simplest and probably
+the most common scenario of creating a library from `.proto` files using the
+cross-platform `dotnet` command. We will implement essentially a clone of the
+`Greeter` library, shared by client and server projects in the [C# `Helloworld`
+example directory
+](https://github.com/grpc/grpc/tree/master/examples/csharp/Helloworld/Greeter).
+
+Start by creating a new library project.
+
+```
+kkm@yupana:~/work$ dotnet new classlib -o MyGreeter
+The template "Class library" was created successfully.
+
+kkm@yupana:~/work$ cd MyGreeter
+kkm@yupana:~/work/MyGreeter$ ls -lF
+total 12
+-rw-rw-r-- 1 kkm kkm   86 Nov  9 16:10 Class1.cs
+-rw-rw-r-- 1 kkm kkm  145 Nov  9 16:10 MyGreeter.csproj
+drwxrwxr-x 2 kkm kkm 4096 Nov  9 16:10 obj/
+```
+
+The `dotnet new` command has created the file `Class1.cs` that we won't need, so
```

Comment:
Nit: it's easier to read if the verb tenses align throughout.  Consider rewording this "The dotnew new command creates the file ...".

Reply:
I rephrased that sentence a bit, but I am not sure I grok your concern. This is as present a tense as it only could be.